### PR TITLE
Add intelliSenseUpdateDelay setting

### PR DESCRIPTION
**Pull request by tjx666**

### CODE REVIEW

**Comment by tjx666** [comment]
@sean-mcmanus #6344

**Comment by michelleangela** [comment]
This requires changes in the closed source part of the code that handles the IntelliSense update.

File: Extension/package.nls.json [hunk 1]
```diff
@@ -134,6 +134,7 @@
...
     "c_cpp.configuration.intelliSenseCacheSize.description": "Maximum size of the per-workspace hard drive space in megabytes (MB) for cached precompiled headers; the actual usage may fluctuate around this value. The default size is 5120 MB. Precompiled header caching is disabled when the size is 0.",
     "c_cpp.configuration.intelliSenseMemoryLimit.description": "Memory usage limit in megabytes (MB) of an IntelliSense process. The default limit is 4096 MB and the maximum limit is 16 GB. The extension will shutdown and restart an IntelliSense process when it exceeds the limit.",
+    "c_cpp.configuration.intelliSenseUpdateDelay.description": "Controls the delay in milliseconds before intelliSense takes effect.",
```
Comment (sean-mcmanus) [inline Extension/package.nls.json]: intelliSense -> IntelliSense

**Comment by sean-mcmanus** [comment]
> This requires changes in the closed source part of the code that handles the IntelliSense update.

Yeah, but it should be pretty trivial to replace the hardcoded value with the setting.

File: Extension/package.json [hunk 2]
```diff
@@ -856,6 +856,12 @@
           "minimum": 256,
           "maximum": 16384
         },
+        "C_Cpp.intelliSenseUpdateDelay": {
+          "type": "number",
+          "default": 2000,
+          "description": "%c_cpp.configuration.intelliSenseUpdateDelay.description%",
+          "scope": "resource"
```
Comment (sean-mcmanus) [inline Extension/package.json]: Could you add
```
          "minimum": 500,
          "maximum": 3000
```

File: Extension/package.nls.json (hunk 1 above)
Comment (sean-mcmanus) [inline Extension/package.nls.json]: "takes effect" -> "starts updating after a modification."

File: Extension/package.json (hunk 2 above)
Comment (sean-mcmanus) [inline Extension/package.json]: I think "resource" should be "application".

File: Extension/src/LanguageServer/client.ts [hunk 3]
```diff
@@ -1184,6 +1186,7 @@ export class DefaultClient implements Client {
                 intelliSenseCachePath : settings_intelliSenseCachePath,
                 intelliSenseCacheSize : settings_intelliSenseCacheSize,
                 intelliSenseMemoryLimit : settings_intelliSenseMemoryLimit,
+                intelliSenseUpdateDelay: settings_intelliSenseUpdateDelay,
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts]: This would be workspaceSettings.intelliSenseUpdateDelay if the scope is changed to "application".

**Comment by sean-mcmanus** [comment]
> > This requires changes in the closed source part of the code that handles the IntelliSense update.
> 
> Yeah, but it should be pretty trivial to replace the hardcoded value with the setting.

I have PR with the change.

**Comment by tjx666** [comment]
all need fixed had done @sean-mcmanus

**Comment by sean-mcmanus** [comment]
> all need fixed had done @sean-mcmanus

Thanks. The setting is working with our binary changes for our pending 1.1.0-insiders3 -- we may have an update later this week.

**Review by sean-mcmanus — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
> [CLA assistant check](https://cla.opensource.microsoft.com/microsoft/vscode-cpptools?pullRequest=6344)
> Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vscode-cpptools?pullRequest=6344) before we can accept your contribution.
> 
> ❌ tjx666 [sign now](https://cla.opensource.microsoft.com/microsoft/vscode-cpptools?pullRequest=6344)
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vscode-cpptools?pullRequest=6344) it.

@tjx666 Can you sign this CLA?

**Comment by tjx666** [comment]
@sean-mcmanus ok

**Pull request merged**